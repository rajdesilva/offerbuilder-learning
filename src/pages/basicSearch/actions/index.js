export { supplySearchActions } from './supplySearch';
export { browseSupplyActions } from './browseSupply';
export { propertyCartActions } from './propertyCart';