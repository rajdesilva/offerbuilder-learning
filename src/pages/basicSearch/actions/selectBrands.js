export const selectBrandsActions = {
    SEARCH_BRANDS_LOADING: 'SEARCH_BRANDS_LOADING',
    SEARCH_BRANDS_SUCCESS: 'SEARCH_BRANDS_SUCCESS',
    SEARCH_BRANDS_FAILURE: 'SEARCH_BRANDS_FAILURE',

    SEARCH_ADD_BRAND: 'SEARCH_ADD_BRAND',
    SEARCH_UPDATE_BRAND: 'SEARCH_UPDATE_BRAND',
    SEARCH_UPDATE_SUPPLIER: 'SEARCH_UPDATE_SUPPLIER',
    SEARCH_UPDATE_STOREFRONT: 'SEARCH_UPDATE_STOREFRONT',
    SEARCH_UPDATE_CHANNEL: 'SEARCH_UPDATE_CHANNEL',
  }