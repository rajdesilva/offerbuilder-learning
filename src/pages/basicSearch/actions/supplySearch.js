export const supplySearchActions = {
  SUPPLY_SEARCH_UPDATE_REDUX_STATE: "SUPPLY_SEARCH_UPDATE_REDUX_STATE",

  SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE: "SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE",
  SUPPLY_SEARCH_UPDATE_BASIC_SETTING_REDUX_STATE: "SUPPLY_SEARCH_UPDATE_BASIC_SETTING_REDUX_STATE",
  SUPPLY_SEARCH_UPDATE_DESTINATION: "SUPPLY_SEARCH_UPDATE_DESTINATION",
  SUPPLY_SEARCH_UPDATE_DISTANCE: "SUPPLY_SEARCH_UPDATE_DISTANCE",
  SUPPLY_SEARCH_UPDATE_HOTEL_NAME: "SUPPLY_SEARCH_UPDATE_HOTEL_NAME",
  SUPPLY_SEARCH_UPDATE_HOTEL_NAMES: "SUPPLY_SEARCH_UPDATE_HOTEL_NAMES",
  SUPPLY_SEARCH_UPDATE_ADDRESS: "SUPPLY_SEARCH_UPDATE_ADDRESS",
  SUPPLY_SEARCH_UPDATE_ZIP: "SUPPLY_SEARCH_UPDATE_ZIP",

  SUPPLY_SEARCH_UPDATE_LOS: "SUPPLY_SEARCH_UPDATE_LOS",
  SUPPLY_SEARCH_UPDATE_SORT_BY_GBL: "SUPPLY_SEARCH_UPDATE_SORT_BY_GBL",
  SUPPLY_SEARCH_UPDATE_PSIZE: "SUPPLY_SEARCH_UPDATE_PSIZE",
  SUPPLY_SEARCH_UPDATE_PNUM: "SUPPLY_SEARCH_UPDATE_PNUM",

  SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE_ALL: "SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE_ALL",
  SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE_ALL_FINAL: "SUPPLY_SEARCH_UPDATE_BASIC_REDUX_STATE_ALL_FINAL",

  SUPPLY_SEARCH_UPDATE_DATE_RANGE: "SUPPLY_SEARCH_UPDATE_DATE_RANGE",
  SUPPLY_SEARCH_ADD_BRAND: "SUPPLY_SEARCH_ADD_BRAND",
  SUPPLY_SEARCH_UPDATE_BRANDS: "SUPPLY_SEARCH_UPDATE_BRANDS",
  SUPPLY_SEARCH_UPDATE_SUPPLIERS: "SUPPLY_SEARCH_UPDATE_SUPPLIERS",
  SUPPLY_SEARCH_UPDATE_CHANNELS: "SUPPLY_SEARCH_UPDATE_CHANNELS",
  SUPPLY_SEARCH_UPDATE_IS_SUPPLIER: "SUPPLY_SEARCH_UPDATE_IS_SUPPLIER",
  SUPPLY_SEARCH_UPDATE_LCN_FLAG: "SUPPLY_SEARCH_UPDATE_LCN_FLAG",
  SUPPLY_SEARCH_UPDATE_REMAINING_CAPITAL_POOL:
    "SUPPLY_SEARCH_UPDATE_REMAINING_CAPITAL_POOL",
  SUPPLY_SEARCH_UPDATE_CURRENCY: "SUPPLY_SEARCH_UPDATE_CURRENCY",
  SUPPLY_SEARCH_UPDATE_SORT_BY: "SUPPLY_SEARCH_UPDATE_SORT_BY",

  SUPPLY_SEARCH_UPDATE_PAGE_OFFSET: "SUPPLY_SEARCH_UPDATE_PAGE_OFFSET",
  SUPPLY_SEARCH_UPDATE_PAGE_SIZE: "SUPPLY_SEARCH_UPDATE_PAGE_SIZE",

  SEARCH_CHANNELS_LOADING: "SEARCH_CHANNELS_LOADING",
  SEARCH_CHANNELS_SUCCESS: "SEARCH_CHANNELS_SUCCESS",
  SEARCH_CHANNELS_FAILURE: "SEARCH_CHANNELS_FAILURE",

  SEARCH_BRANDS_LOADING: "SEARCH_BRANDS_LOADING",
  SEARCH_BRANDS_SUCCESS: "SEARCH_BRANDS_SUCCESS",
  SEARCH_BRANDS_FAILURE: "SEARCH_BRANDS_FAILURE",

  SEARCH_ADD_BRAND: "SEARCH_ADD_BRAND",
  SEARCH_UPDATE_BRAND: "SEARCH_UPDATE_BRAND",
  SEARCH_REMOVE_BRAND: "SEARCH_REMOVE_BRAND",

  SEARCH_UPDATE_SUPPLIER: "SEARCH_UPDATE_SUPPLIER",
  SEARCH_REMOVE_SUPPLIER: "SEARCH_REMOVE_SUPPLIER",

  SEARCH_UPDATE_STOREFRONT: "SEARCH_UPDATE_STOREFRONT",
  SEARCH_REMOVE_STOREFRONT: "SEARCH_REMOVE_STOREFRONT",
  SEARCH_ADD_ROW_STOREFRONT: "SEARCH_ADD_ROW_STOREFRONT",

  SEARCH_UPDATE_CHANNEL: "SEARCH_UPDATE_CHANNEL",
  SEARCH_REMOVE_CHANNEL: "SEARCH_REMOVE_CHANNEL",

  RESET_SEARCH_PARAM_STATE: "RESET_SEARCH_PARAM_STATE",

  FETCH_CURRENCIES_LOADING: "FETCH_CURRENCIES_LOADING",
  FETCH_CURRENCIES_SUCCESS: "FETCH_CURRENCIES_SUCCESS",
  FETCH_CURRENCIES_FAILURE: "FETCH_CURRENCIES_FAILURE",

  SET_SELECTED_CURRENCY: "SET_SELECTED_CURRENCY",

  FETCH_MARKET_PRICE_LOADING: "FETCH_MARKET_PRICE_LOADING",
  FETCH_MARKET_PRICE_SUCCESS: "FETCH_MARKET_PRICE_SUCCESS",
  FETCH_MARKET_PRICE_FAILURE: "FETCH_MARKET_PRICE_FAILURE",
  SET_INCLUDE_MARKET_PRICE_FLAG: "SET_INCLUDE_MARKET_PRICE_FLAG",

  SUPPLY_SEARCH_UPDATE_PROPERTY_TYPE: "SUPPLY_SEARCH_UPDATE_PROPERTY_TYPE",
};
