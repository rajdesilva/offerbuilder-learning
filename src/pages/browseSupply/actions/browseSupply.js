export const browseSupplyActions = {
  BROWSE_SUPPLY_SEARCH_LOADING: 'BROWSE_SUPPLY_SEARCH_LOADING',
  BROWSE_SUPPLY_SEARCH_SUCCESS: 'BROWSE_SUPPLY_SEARCH_SUCCESS',
  BROWSE_SUPPLY_SEARCH_FAILURE: 'BROWSE_SUPPLY_SEARCH_FAILURE',

  ADD_TEMP_EDITED_BRANDS: "ADD_TEMP_EDITED_BRANDS",
  REMOVE_TEMP_EDITED_BRANDS: "REMOVE_TEMP_EDITED_BRANDS",
  RESET_TEMP_EDITED_BRANDS: "RESET_TEMP_EDITED_BRANDS",
  RESET_SEARCHED_PROPERTIES: "RESET_SEARCHED_PROPERTIES",
}