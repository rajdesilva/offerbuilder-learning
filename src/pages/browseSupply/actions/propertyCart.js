export const propertyCartActions = {
    ADD_PROPERTY_TO_CART: 'ADD_PROPERTY_TO_CART',
    DELETE_CART_ITEM: 'DELETE_CART_ITEM',
    EMPTY_CART: 'EMPTY_CART',
    UPDATE_ENTIRE_CART_LIST: 'UPDATE_ENTIRE_CART_LIST',
    DISCARD_CHANGES_IN_EDITED_CART_LIST: "DISCARD_CHANGES_IN_EDITED_CART_LIST",
    UPDATE_LANGUAGE_WITH_DESCRIPTION_IN_PROPERTY: 'UPDATE_LANGUAGE_WITH_DESCRIPTION_IN_PROPERTY',

    UPDATE_SHORT_DESCRIPTION_FOR_CART_ITEM: "UPDATE_SHORT_DESCRIPTION_FOR_CART_ITEM",
    UPDATE_DESCRIPTION_FOR_CART_ITEM: "UPDATE_DESCRIPTION_FOR_CART_ITEM",
    UPDATE_IMAGES_FOR_CART_ITEM: "UPDATE_IMAGES_FOR_CART_ITEM",
    UPDATE_CART_CHANGE_STATUS: "UPDATE_CART_CHANGE_STATUS",
  }