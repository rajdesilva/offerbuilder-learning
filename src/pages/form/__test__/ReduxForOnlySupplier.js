export const ReduxForOnlySupplier = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [
        {
          id: "ntp",
          name: "NTP",
        },
      ],
      channels: [
        {
          id: "AT_CORDIAL",
          name: "",
        },
      ],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
