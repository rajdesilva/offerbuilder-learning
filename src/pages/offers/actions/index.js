export { newOfferActions } from './newOfferActions';
export { marketingActions } from './marketingActions';
export { offerListActions } from './offerListActions';
export { offerListSearchAndFilterActions } from './offerListSearchAndFilterActions';