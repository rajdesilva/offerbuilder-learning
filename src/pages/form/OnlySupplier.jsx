import React, { Fragment } from "react";
import { Row, Col, Form, Spin } from "antd";
import { useDispatch, useSelector } from "react-redux";
import NemoSelect from "./NemoSelect";
import { FormattedMessage } from "react-intl";
import { supplySearchActions } from "../browseSupply/actions";
import styles from "./css/OnlySupplier.module.less";

function OnlySupplier(props) {
  const dispatch = useDispatch();
  const { channels, loading } = useSelector((state) => ({
    channels: window.getValue(state, "channelinfo.channels"),
    loading: window.getValue(state, "channelinfo.loading"),
  }));
  return loading ? (
    <Row gutter={[8, 8]}>
      <Col span={4} className="col1-text"></Col>
      <Spin className={styles["brand-spinner"]} />
    </Row>
  ) : channels.length === 0 ? (
    <Row gutter={[8, 8]}>
      <Col span={4} className="col1-text"></Col>
      <FormattedMessage id="nemo.noChannelsFound" />
    </Row>
  ) : (
    <Row gutter={[8, 8]}>
      {props.isForSearchResultsPage ? (
        <Fragment />
      ) : (
        <Col span={4} className="col1-text"></Col>
      )}
      <Col span={props.isForSearchResultsPage ? 10 : 6}>
        <Form.Item
          name={["target", "suppliers"]}
          label={<FormattedMessage id="nemo.supplier" />}
          rules={[
            {
              required: true,
              message: (
                <FormattedMessage id="nemo.pleaseSelectAtleastOneSupplier" />
              ),
            },
          ]}
          className={styles["brand-config-channel"]}
          initialValue={props.form.getFieldValue(["suppliers"])}
        >
          <NemoSelect
            data-testid="supplier-select"
            mode="multiple"
            id="brw-search-supply"
            optionsList={[
              {
                id: "ntp",
                name: "NTP",
              },
            ]}
            onChange={(value) => {
              dispatch({
                type: supplySearchActions.SUPPLY_SEARCH_UPDATE_SUPPLIERS,
                payload: value.map((channel) => JSON.parse(channel)),
              });
            }}
            defaultValue={props.form.getFieldValue(["suppliers"])}
          />
        </Form.Item>
      </Col>
      <Col span={props.isForSearchResultsPage ? 14 : 7}>
        <Form.Item
          name={["target", "channels"]}
          label={<FormattedMessage id="nemo.channelsOnlyForNtp" />}
          rules={[
            {
              required: true,
              message: (
                <FormattedMessage id="nemo.pleaseSelectAtleastOneChannels" />
              ),
            },
          ]}
          className={styles["onlysupplier-channel"]}
          initialValue={props.form.getFieldValue([
            "suppliers",
            "0",
            "channels",
          ])}
        >
          <NemoSelect
            data-testid="channels-select"
            mode="multiple"
            id="brw-search-channel"
            maxTagCount={2}
            optionsList={channels}
            onChange={(value) => {
              dispatch({
                type: supplySearchActions.SUPPLY_SEARCH_UPDATE_CHANNELS,
                payload: value.map((channel) => JSON.parse(channel)),
              });
            }}
            defaultValue={props.form.getFieldValue([
              "suppliers",
              "0",
              "channels",
            ])}
          />
        </Form.Item>
      </Col>
    </Row>
  );
}

export default OnlySupplier;
