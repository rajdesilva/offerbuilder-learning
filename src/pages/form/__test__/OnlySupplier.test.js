import React from "react";
import OnlySupplier from "../OnlySupplier";
import { ReduxForOnlySupplier } from "./ReduxForOnlySupplier";
import { createStore } from "redux";
import { Form } from "antd";

import { act, render, fireEvent, cleanup } from "../../../helpers/testUtils";
import { supplySearchActions } from "../../browseSupply/actions";

afterEach(cleanup);

describe("OnlySupplier component test cases", () => {
  test("OnlySupplier components snapshot", () => {
    const props = {};
    const store = createStore(() => ({ ...ReduxForOnlySupplier }));

    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <Form form={form} initialValues={{}}>
          <OnlySupplier {...{ ...props, form }} />
        </Form>
      );
    };

    const { container } = render(<FormWrapper {...props} />, {
      store,
    });

    expect(container.firstChild).toMatchSnapshot();
  });
  test("render onlysupplier component having nothing pass in the store and nothing needs to selected in the dom ", () => {
    const props = {};
    const store = createStore(() => ({ ...ReduxForOnlySupplier }));

    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <Form form={form} initialValues={{}}>
          <OnlySupplier {...{ ...props, form }} />
        </Form>
      );
    };
    const { queryAllByText } = render(<FormWrapper {...props} />, {
      store,
    });
    expect(queryAllByText("Please Select")).toHaveLength(2);
  });
  test("display the value from store to component for supplier and channels ", () => {
    const props = {};
    const store = createStore(() => ({ ...ReduxForOnlySupplier }));

    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <Form form={form} initialValues={ReduxForOnlySupplier.searchparams}>
          <OnlySupplier {...{ ...props, form }} />
        </Form>
      );
    };
    const { getByText } = render(<FormWrapper {...props} />, {
      store,
    });
    expect(
      getByText(ReduxForOnlySupplier.searchparams.target.suppliers[0].name)
    ).toBeInTheDocument();
    expect(
      getByText(ReduxForOnlySupplier.searchparams.target.channels[0].id)
    ).toBeInTheDocument();
  });
  test("add more values to channels and verify it ", async () => {
    const props = {};
    const store = createStore(() => ({ ...ReduxForOnlySupplier }));
    store.dispatch = jest.fn();
    const FormWrapper = (props) => {
      const [form] = Form.useForm();
      return (
        <Form form={form} initialValues={ReduxForOnlySupplier.searchparams}>
          <OnlySupplier {...{ ...props, form }} />
        </Form>
      );
    };
    const { getByTestId, queryAllByRole } = render(<FormWrapper {...props} />, {
      store,
    });
    const select = queryAllByRole("combobox");
    await act(async () => {
      await fireEvent.keyDown(select[select.length - 1], { key: "ArrowDown" });
      await fireEvent.click(
        getByTestId(
          "nemo-select-brw-search-channel-option-" +
            ReduxForOnlySupplier.channelinfo.channels[1].id
        )
      );
    });

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith({
      type: supplySearchActions.SUPPLY_SEARCH_UPDATE_CHANNELS,
      payload: ReduxForOnlySupplier.channelinfo.channels.map((c) => ({
        id: c.id,
        name: c.name,
      })),
    });
  });
});
