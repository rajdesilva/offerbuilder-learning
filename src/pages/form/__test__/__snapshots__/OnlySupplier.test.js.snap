// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlySupplier component test cases OnlySupplier components snapshot 1`] = `
<form
  class="ant-form ant-form-horizontal"
>
  <div
    class="ant-row"
    style="margin: -4px -4px 4px -4px;"
  >
    <div
      class="ant-col ant-col-4 col1-text"
      style="padding: 4px 4px 4px 4px;"
    />
    <div
      class="ant-col ant-col-6"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="target_suppliers"
            title=""
          >
            Supplier
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-multiple ant-select-show-search"
                data-testid="nemo-select-brw-search-supply"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                    style="width: 0px;"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="ant-select-selection-search-mirror"
                    >
                      
                       
                    </span>
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    Please Select
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-7"
      style="padding: 4px 4px 4px 4px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="target_channels"
            title=""
          >
            Channels Only For NTP
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-select ant-select-multiple ant-select-show-search"
                data-testid="nemo-select-brw-search-channel"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                    style="width: 0px;"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="ant-select-selection-search-mirror"
                    >
                      
                       
                    </span>
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    Please Select
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
